### PR TITLE
EG-422, Create Url, Controller, Service for admin home page

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/models/Enumerable.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import play.api.libs.json._
+
+trait Enumerable[A] {
+
+  def withName(str: String): Option[A]
+}
+
+object Enumerable {
+
+  def apply[A](entries: (String, A)*): Enumerable[A] =
```
Comment (LesWilson) [inline app/models/Enumerable.scala:28]: This method can be changed to 
`  def apply[A](entries: (String, A)*): Enumerable[A] =
    (str: String) => entries.toMap.get(str)
`

File: test/models/ApprovalStatusSpec.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+package models
+
+import org.scalatest.{MustMatchers, OptionValues, WordSpec}
+import play.api.libs.json.{JsError, Json, JsString, JsObject, JsNull}
+import models.ApprovalStatus._
+
+class ApprovalStatusSpec extends WordSpec with MustMatchers with OptionValues with Enumerable.Implicits {
+
+  "CompanyApplyingIsRightsHolder" must {
```
Comment (LesWilson) [inline test/models/ApprovalStatusSpec.scala:25]: Minor: the description of the "must" doesn't tie in to the test

File: app/models/Enumerable.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/models/Enumerable.scala:28]: but less clear.  Removing the return type, the suggestion fails to compile, whereas the original implementation compiles and runs as intended

File: test/models/ApprovalStatusSpec.scala (hunk 2 above)
Comment (cmrhmrc) [reply test/models/ApprovalStatusSpec.scala:25]: ha yes, will update

**Pull request merged**